TASK: Extract methods `set`, `get`, `getByTag`, `has`, and `remove` from `FrontendInterface` into `FlowCacheFrontendInterface`

mficzel opened the pull request:
The frontend interface now only describes method needed by the cache manager and the methods used for interaction with the cache are moved to the new `FlowCacheFrontendInterface`. This change will allow to implement frontends that provide a differrent api like PSR6, PSR16 directly.

About breakiness:
- Since code always use one of the concrete frontend implementations no changed are needed
- Cache Backends still get the same Frontend set and as long as the backends only use the getIdentifier
method which is common no changes are needed.

**Review instructions**

Stumbled upon this when i played with https://github.com/sitegeist/Sitegeist.StaleMate (Caches with Stale items and async udates) and was not able to implement those in an alternate CacheFrontend. Also this will allow to create PSR6, PSR16  compatible cache frontends that are handled like all the other frontends and 

**Checklist**

- [x] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked wit `!!!` and have upgrade-instructions

mficzel commented:
I wonder wether the constructors should be part of the cache interface since the cache manager relies on those for instantiating. Not sure wether this would be breaking as caches have implement them implicitly anyways.

kitsunet commented:
Definitely breaking as any existing Frontend would not be implementing the right interface anymore. Will look more into it later. But generally like the separation

mficzel commented:
@kitsunet I adjusted to extract a LowLevelCacheInterface and leave the FrontendInterface as it is ... not sure wether this is less breaky or a good idea.

mficzel committed:
TASK: Move methods `set`, `get`, `getByTag`, `has`, and `remove` from `FrontendInterface` into new `FlowCacheFrontendInterface`

The frontend interface now only describes method needed by the cache manager and the methods used for interaction with the cache
are moved to the new `FlowCacheFrontendInterface`. This change will allow to implement frontends that provide a differrent api like PSR6, PSR16
directly.

About breakiness:
- Since code always use one of the concrete frontend implementations no changed are needed
- Cache Backends still get the same Frontend set and as long as the backends only use the getIdentifier
  method which is common no changes are needed.

mficzel commented:
I reverted the idea of a LowLevelCacheInterface since this would require changes in the backend implementations (other frontend is passed) and am back to the `FlowCacheFrontendInterface ` on top of `FrontendInterface`.

mficzel commented:
To discuss:
- Which methods should be extracted from the frontend interface
- Is there a better name then FlowCacheFrontendInterface

mficzel commented:
Close for now as this is not finished and will need a rework afterwards
